content/java/concepts/errors/terms/invalidclassexception/invalidclassexception.md

```diff
@@ -0,0 +1,21 @@
+---
+Title: 'InvalidClassException'
+Description: 'Occurs when the Serialization runtime detects a problem with a class.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Errors'
+  - 'Error Handling'
+  - 'Exceptions'
+  - 'Debugging'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`InvalidClassException`** is an [exception](https://www.codecademy.com/resources/docs/general/error) that occurs when the Serialization runtime detects a problem with a [class](https://www.codecademy.com/resources/docs/java/classes). The problem can be one of the following:
+
+- The serial version of the class does not match that of the class descriptor read from the stream.
+- The class contains unknown datatypes.
+- The class does not have an accessible constructor without arguments.
```

Comment:
```suggestion
- The class does not have an accessible [constructor](https://www.codecademy.com/resources/docs/java/constructors) without arguments.
```

Reply:
Could/should we also include a brief example for this Exception?

Reply:
@Dusch4593, It's one of those errors that it's hard to come up with a straightforward example.